manager/apis/app/v1alpha1/m4dapplication_types.go

```diff
@@ -75,6 +82,12 @@ type DataContext struct {
 	// Default is none
 	// +optional
 	Recurrence RecurrenceType `json:"recurrence,omitempty"`
+
+	// Cataloged indicates whether or not the data set has already been registered in the organization's data catalog.
+	// When a user chooses the data set from the catalog this should be true.  In flows where a new data set is
+	// being created in the M4D managed environment this should be false.
+	// +optional
+	Cataloged bool `json:"cataloged,omitempty"`
```

Comment:
Do we specify both an external catalog id and the state of the asset? If data comes from an external catalog, then it is not cataloged in the organization catalog.
Another question: supporting multiple catalogs - what does it mean? If we support multiple catalogs then we can also catalog the asset in more than one catalog.

Reply:
@shlomitk1 Not sure what you mean by "state of the asset".
Regarding multiple catalogs, I will let @roee88 answer.  I think for the ingest scenario we will assume two catalogs.  One is the "inline" catalog via which the data sets to be ingested are provided, and the other is the enterprise catalog where the data sets are registered after they are successfully ingested into the data lake.

@roee88 Should the data sets be removed from the "inline" catalog after they are registered in the enterprise catalog?

Reply:
@simanadler I mean that if dataCatalogId=inline, you don't need the 'cataloged' field. 
My concern is the user experience. Along with specifying the data identificators, you also need to explicitly state that it is cataloged, since a dataset is not cataloged by the default. I would like to remove the need in this property entirely, or to change the default (cataloged -> external)